Documentation/watch-recovery.md

```diff
@@ -0,0 +1,104 @@
+```
+
+Remember that the kubelet (the Kubernetes node agent) runs on reconciliation loops which work to keep the system in a stable state. When we execute the command to kill this API server, the kubelet will notice immediately.
+
+Test this by killing the active containers running the Kubernetes API server on this node. Because the API server is a DaemonSet, it should be running on all of the master nodes.
+
+From one of the masters, get the IDs of the containers running the API Server:
+
+```
+$ docker ps -f 'name=.*kube-apiserver.*' --format "{{.ID}}"
+4ae686927f22
+06bb5ba95034
+```
+Then, kill the listed containers.
+
+```
+$ docker kill 4ae686927f22 06bb5ba95034
+```
+
+Within a few seconds the kubelet will notice that the API server Pod is no longer running, and will relaunch under a new ID:
+
+```
+$ docker ps -f 'name=.*kube-apiserver.*' --format "{{.ID}}"
+f4a262619d1d
+d964fa94c69b
+```
+
+In the Console, watch the list of API servers to see that one is disabled, and a new one is launched. Note that the Tectonic Console session offers continuous feedback because the API is configured with High Availability. This is the same Kubernetes feature that enables your apps to remain up and running without constant developer monitoring and interaction.
+
+We just proved that our cluster can not only recover from a single Pod failing, but it can handle failure of the most critical type, it's own API server.
+
+## Simulate Node failure
+
+Killing the API server acts as an example of an individual process failure. Next, simulate a power down or network partition by disabling an entire node.
+
+Kubelet uses reconciliation loops to remain in constant contact with the API server and report back status. Use iptables to create a temporary firewall rule and simulate an interruption in this connection.
+
+```
+$ iptables -A OUTPUT -p tcp --dport 443  -j DROP
+$ systemctl stop dockerk
+$ systemctl stop kubelet
+$ ps aux | grep kube
+```
+
+Use Tectonic Console to inspect the node. Notice that the node is marked ‘unhealthy’, but nothing else has changed about workloads.
+
+Check your app. Even though the node is unhealthy, and the rescheduling process has not yet begun, the app is still up and running as before. Because the cluster runs in High Availability, the app does not go down, and your clients experience no interruption in service. Kubernetes maintains persistent availability automatically.
+
+To ensure that workloads aren't constantly shuffled around for a minor networking outage, the kubelet will wait 5 minutes for transient errors to clear, before taking more drastic action. After 5 minutes, reinspect Console. Notice that new Pods have been started and scheduled to repair the interruption.
+
+Undo the temporary firewall rule to watch the node recover almost immediately.
+
+```
+$ iptables -A OUTPUT -p tcp --dport 443  -j ACCEPT
+$ systemctl start dockerk
+$ systemctl enable kubelet
+$ ps aux | grep kube
+
+```
+
+## More testing opportunities
```

Comment:
@robszumski Without links to relevant doc, it doesn't make sense to add this section. I cannot find Tectonic doc to fulfill these three bullets.

could we go:
* performance and load testing (?)
* use Deployments to scale an app (https://coreos.com/tectonic/docs/latest/tutorials/sandbox/scale-app.html)
* review resource limits and namespace quotas (https://coreos.com/tectonic/docs/latest/tutorials/sandbox/check-logs.html ?)

Reply:
Link to upstream? My thinking was that it's a hint that there is much more to do that just this guide. If the mindset of someone landing here is to learn, here are more things you can learn about.

https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/
https://kubernetes.io/docs/tasks/administer-cluster/memory-default-namespace/
https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/